test/slack/message_test.exs

```diff
@@ -0,0 +1,110 @@
+
+defmodule Slack.MessageTest do
+  use ExUnit.Case
+  alias Slack.{Message,Channel,FakeWebApi}
+
+  setup do
+    FakeWebApi.start_link
+
+    :ok
+  end
+
+  describe "new/1" do
+    test "returns message struct" do
+      assert Message == Message.new(channel: channel(), sender: user(), ts: "1355517523.000005", text: "My message").__struct__
+    end
+
+    test "handles different arg orders" do
+      assert Message.new(text: "My message", channel: channel(), ts: "1355517523.000005", sender: user())
+    end
+
+    test "raises exception on missing options" do
+      assert_raise ArgumentError, fn -> Message.new(text: "My message") end
+    end
+  end
+
+
+  describe "new_from_event/1" do
+    test "returns message struct" do
+      message_event = %{
+        channel: "Ctestchannel",
+        user: "Utestuser",
+        ts: "1355517523.000005",
+        text: "My message"
+      }
+
+      assert message? Message.new_from_event(slack(), message_event)
+    end
+  end
+
+  describe "permalink/2" do
+    test "returns url" do
+      msg = Message.new(channel: channel(), sender: user(), ts: "1355517523.000005", text: "My message")
+
+      assert "http://example.com/archives/Ctestchannel/1355517523.000005" ==
+        Message.permalink(slack(), msg)
+    end
+
+    test "calls chat.getPermalink api method" do
+      msg = Message.new(channel: channel(), sender: user(), ts: "1355517523.000005", text: "My message")
+      Message.permalink(slack(), msg)
+
+      assert api_method_called("chat.getPermalink")
+    end
+
+    test "passes message timestamp argument" do
+      msg = Message.new(channel: channel(), sender: user(), ts: "1355517523.000005", text: "My message")
+      Message.permalink(slack(), msg)
+
+      assert passed_api_argument(:message_ts, "1355517523.000005")
+    end
+
+    test "passes channel argument" do
+      msg = Message.new(channel: channel(), sender: user(), ts: "1355517523.000005", text: "My message")
+      Message.permalink(slack(), msg)
+
+      assert passed_api_argument(:channel, "Ctestchannel")
+    end
+
+  end
+
+  # Background & support
+  # ----
+
+  defp channel() do
+    Channel.new(id: "Ctestchannel", name: "Test channel")
+  end
+  defp user(), do:  nil
```

Comment:
I prefer to give it a name so that the reader of the test knows what is *suppose* to be passed there. I figure at some point we will need real user there anyway.